### PR TITLE
Iss 54

### PR DESCRIPTION
**Pull request by jedpittman**

### CODE REVIEW

**Comment by jedpittman** [comment]
with this commit, there should be a working api for states (including data/migration): http://127.0.0.1:8000/jobs/state/

File: jobs/urls.py [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 from . import views
 from rest_framework import routers
-from .api import LeadViewSet, BlsOesViewSet
+from .api import LeadViewSet, BlsOesViewSet, StateViewSet
 
 router = routers.DefaultRouter()
 router.register('leads', LeadViewSet)
 router.register('soc-codes', BlsOesViewSet)
+router.register(r'state', StateViewSet, basename='abbr')
+# router.register(r'state(?P<state_abb>.+)/$', StateViewSet, basename='state-name')
```
Comment (alexjball) [inline jobs/urls.py:10]: remove comment

File: References/DjangoData.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Django is a python based framework that helps us provide standard rest apis to our database backend in postgres. Using a framework allows us to develop solid api endpoints for our application with less time. This document is written from windows for developers of the jobhopper application. Users should be familiar with python, postgres, virtual env and have gotten the application running. This doc explains more detail about how to change the model/data/database structure.
+
+In order to keep our django application in sync with our database, we leverage Django migrations. Migrations are generally run with the command similar to this (windows):
+
+Migrations can update either the database schema (the tables/structure) or the data inside. Both types of migrations are saved in the migrations subfolder.
+
+## See Migrations and which have been applied:
+
+(venv) C:\Users\jedpi\OneDrive\Documents\GitHub\jobhopper>python manage.py showmigrations
```
Comment (alexjball) [inline References/DjangoData.md:11]: remove your shell prefix, so it's only the command. And wrap in `` so it renders as code

File: jobs/migrations/0008_auto_20201006_AddStateData.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
+# Generated by Django 3.1 on 2020-10-07 01:42
+from __future__ import unicode_literals
+from django.db import migrations, models
+
+def add_states(apps, schema_editor):
```
Comment (alexjball) [inline jobs/migrations/0008_auto_20201006_AddStateData.py:5]: Do these match the `area_title` values in the BLS OES dataset? I think they need to match so we can look things up. Do we also need a national code? I'm not sure what the BLS dataset uses for that.

File: jobs/test_api_soc.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# Create your tests here.
+from django.urls import reverse
+
+
+class SocCodesAPITests(TestCase):
+    def test_soccodes_urls_return_200(self):
+        """
+        Make sure that url returns 200 status code
+        """
+        # True at the end makes it follow.
```
Comment (alexjball) [inline jobs/test_api_soc.py:12]: What does this comment mean? is it necessary?

File: jobs/test_api_states.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+class StatesAPITests(TestCase):
+    def test_state_urls_return_200(self):
+        """
+        Make sure that url returns 200 status code
+        """
+        # True at the end makes it follow.
+        response = self.client.get("/jobs/state", {}, True)
+        # print(dir(response))
+        # print(f"result code is {response.status_code}")
```
Comment (alexjball) [inline jobs/test_api_states.py:15]: remove comments

File: jobs/test_api_states.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+# Create your tests here.
+from django.urls import reverse
+
+
+class StatesAPITests(TestCase):
+    def test_state_urls_return_200(self):
+        """
+        Make sure that url returns 200 status code
+        """
+        # True at the end makes it follow.
```
Comment (alexjball) [inline jobs/test_api_states.py:12]: what does this mean?

**Review by alexjball — COMMENTED** [review]
Were you able to run tests? They seem to pass, and I can reach the endpoint -- great work! See my question about area_title

```
test_urls_return_200 (jobhopper.test_api_healthcheck.HealthCheckAPITests)
Make sure that url returns 200 status code ... ok
test_urls_return_valid_code (jobhopper.test_api_leads.LeadsApiTests)
Make sure that url returns valid status code ... ok
test_soccodes_urls_return_200 (jobs.test_api_soc.SocCodesAPITests)
Make sure that url returns 200 status code ... ok
test_state_urls_return_200 (jobs.test_api_states.StatesAPITests)
Make sure that url returns 200 status code ... ok
```

File: jobs/test_api_soc.py (hunk 4 above)
Comment (jedpittman) [reply jobs/test_api_soc.py:12]: removed. unnecessary.

File: jobs/migrations/0008_auto_20201006_AddStateData.py (hunk 3 above)
Comment (jedpittman) [reply jobs/migrations/0008_auto_20201006_AddStateData.py:5]: Added issue #122 to handle this. It was not in the original scope but I agree we should be working towards it.

**Pull request merged**